csharp/src/Google.Protobuf/CodedInputStream.cs

```diff
@@ -334,49 +366,55 @@ public uint PeekTag()
         /// for an embedded message, for example.
         /// </remarks>
         /// <returns>The next field tag, or 0 for end of stream. (0 is never a valid tag.)</returns>
+        [SecuritySafeCritical]
         public uint ReadTag()
         {
-            if (hasNextTag)
-            {
-                lastTag = nextTag;
-                hasNextTag = false;
-                return lastTag;
-            }
+            var immediateBudder = ImmediateBuffer;

```

Comment:
nit: is there a precedent for using "budder" as a buffer?   I know about Java's clazz being used instead of class, but "budder" feels a bit random.

Reply:
This is a typo - sorry for that